MimeKit/Cryptography/AsymmetricAlgorithmExtensions.cs

```diff
@@ -247,8 +246,15 @@ static AsymmetricAlgorithm GetAsymmetricAlgorithm (DsaPrivateKeyParameters key,
 			var parameters = GetDSAParameters (key);
 			parameters.X = GetPaddedByteArray (key.X, parameters.Q.Length);
 
-			if (pub != null)
+			if (pub != null) {
 				parameters.Y = GetPaddedByteArray (pub.Y, parameters.P.Length);
+			
+			}
+			else
+			{
+				// If pub is null, derive Y from the private key parameters
+				parameters.Y = key.Parameters.G.ModPow(key.X, key.Parameters.P).ToByteArrayUnsigned ();
+			}
```

Comment:
This change was initiated when I saw this Test failure in the build logs:
https://github.com/jstedfast/MimeKit/actions/runs/12252099258/job/34178015357#annotation:11:73

```
1) Error : UnitTests.Cryptography.AsymmetricAlgorithmExtensionTests.TestDSACryptoServiceProvider
System.ArgumentException : The specified DSA parameters are not valid; P, Q, G and Y are all required.
   at System.Security.Cryptography.DSAOpenSsl.ImportParameters(DSAParameters parameters)
   at System.Security.Cryptography.DSACryptoServiceProvider.ImportParameters(DSAParameters parameters)
   at MimeKit.Cryptography.AsymmetricAlgorithmExtensions.GetAsymmetricAlgorithm(DsaPrivateKeyParameters key, DsaPublicKeyParameters pub) in /home/runner/work/MimeKit/MimeKit/MimeKit/Cryptography/AsymmetricAlgorithmExtensions.cs:line 255
   at MimeKit.Cryptography.AsymmetricAlgorithmExtensions.AsAsymmetricAlgorithm(AsymmetricKeyParameter key) in /home/runner/work/MimeKit/MimeKit/MimeKit/Cryptography/AsymmetricAlgorithmExtensions.cs:line 333
   at UnitTests.Cryptography.AsymmetricAlgorithmExtensionTests.AssertDSA(DSA dsa) in /home/runner/work/MimeKit/MimeKit/UnitTests/Cryptography/AsymmetricAlgorithmExtensionTests.cs:line 73
   at UnitTests.Cryptography.AsymmetricAlgorithmExtensionTests.TestDSACryptoServiceProvider() in /home/runner/work/MimeKit/MimeKit/UnitTests/Cryptography/AsymmetricAlgorithmExtensionTests.cs:line 125
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

Reply:
Yea, that's been an intermittent failure for years - I just hadn't taken the time to debug it. So thanks for taking the time :-)